contracts/dtoken/src/common.rs

```diff
@@ -48,13 +48,34 @@ impl Contract {
         self.underlying_token.clone()
     }
 
-    pub fn get_exchange_rate(&self, underlying_balance: WBalance) -> Ratio {
-        if self.token.total_supply == 0 {
+    pub fn get_exchange_rate(
+        &self,
+        underlying_balance: WBalance,
+        total_borrows: Balance,
+        total_reserves: Balance,
+        total_supplies: Balance,
+    ) -> Ratio {
+        self.calculate_exchange_rate(
+            underlying_balance,
+            total_borrows,
+            total_reserves,
+            total_supplies,
+        )
+    }
+
+    fn calculate_exchange_rate(
+        &self,
+        underlying_balance: WBalance,
+        total_borrows: Balance,
+        total_reserves: Balance,
+        total_supplies: Balance,
+    ) -> Ratio {
+        // initial exchange rate with 0 supply pointed in config while deploying contract
+        if total_supplies == 0 {
             return self.initial_exchange_rate;
```

Comment:
as we can have different initial_exchange rate, i think its more convenient to read from contract, but if we would have initial_exchange_rate = 1 all the time, i can make in a const variable. What way is better. @siexp